### PR TITLE
Fix or disable compilation warnings

### PR DESCRIPTION
**Pull request by jacek-manko-red**
Most of warnings have been fixed, some were disabled:
-Wformat=1 (2 enables -Wformat-nonliteral, nonliteral format strings are used in Angle)
-Wcast-align - disabled, it generates many warnings in e.g Angle, xdgmime or JavaScriptCore
-Wpragmas - disabled (macros for disabling case fallthrough warnings fail in DerivedSources)

It should now be possible to compile with -Werror.
I can't check every relevant platform and configuration though, so this commit does not enable -Werror

### CODE REVIEW

File: Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp [hunk 1]
```diff
@@ -240,6 +240,8 @@ class GstVideoFrameHolder : public TextureMapperPlatformLayerBuffer::UnmanagedBu
 public:
     explicit GstVideoFrameHolder(GstSample* sample, TextureMapperGL::Flags flags, bool gstGLEnabled)
     {
+        UNUSED_PARAM(flags);
```
Comment (magomez) [inline Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp:244]: These should be marked as unused only if GSTREAMER_GL is not used.

File: Source/WebCore/rendering/RenderLayerCompositor.cpp [hunk 2]
```diff
@@ -759,6 +759,8 @@ bool RenderLayerCompositor::updateCompositingLayers(CompositingUpdateType update
 
         auto& frame = m_renderView.frameView().frame();
         bool isMainFrame = isMainFrameCompositor();
+        UNUSED_PARAM(frame);
```
Comment (magomez) [inline Source/WebCore/rendering/RenderLayerCompositor.cpp:763]: These vars are used by the LOG_WITH_STREAM macro, which doesn't to anything when log is disabled. The parameters should be marked as unused if log is disabled.

File: Source/WebCore/rendering/RenderLayerCompositor.cpp [hunk 3]
```diff
@@ -791,6 +793,7 @@ bool RenderLayerCompositor::updateCompositingLayers(CompositingUpdateType update
 #if !LOG_DISABLED || ENABLE(RESOURCE_USAGE)
     if (compositingLogEnabled() && isFullUpdate && (needHierarchyUpdate || needGeometryUpdate)) {
         MonotonicTime endTime = MonotonicTime::now();
+        UNUSED_PARAM(endTime);
```
Comment (magomez) [inline Source/WebCore/rendering/RenderLayerCompositor.cpp:799]: Same case as the previous frame and isMainFrame vars.

File: Source/WebKit/StorageProcess/StorageProcess.cpp [hunk 4]
```diff
@@ -310,11 +310,13 @@ void StorageProcess::fetchWebsiteData(PAL::SessionID sessionID, OptionSet<Websit
 {
     auto websiteData = std::make_unique<WebsiteData>();
     WebsiteData* rawWebsiteData = websiteData.get();
+    UNUSED_PARAM(rawWebsiteData);
     auto callbackAggregator = CallbackAggregator::create([this, websiteData = WTFMove(websiteData), callbackID]() {
         parentProcessConnection()->send(Messages::StorageProcessProxy::DidFetchWebsiteData(callbackID, *websiteData), 0);
     });
 
     String path;
+    UNUSED_PARAM(path);
```
Comment (magomez) [inline Source/WebKit/StorageProcess/StorageProcess.cpp]: Both rawWebsiteData and path are used when SERVICE_WORKER or INDEXED_DATABASE are enabled, so they should only be marked as unused in the rest of the cases.

File: Tools/wpe/backends/WindowViewBackend.cpp [hunk 5]
```diff
@@ -378,6 +378,10 @@ const struct wl_touch_listener WindowViewBackend::s_touchListener = {
     [](void*, struct wl_touch*) { },
     // cancel
     [](void*, struct wl_touch*) { },
+    // shape
+    [](void *, struct wl_touch*, int32_t, wl_fixed_t, wl_fixed_t) { },
+    // orientation
+    [](void *, struct wl_touch*, int32_t, wl_fixed_t) { },
```
Comment (magomez) [inline Tools/wpe/backends/WindowViewBackend.cpp:385]: These two functions structure members are present only when wayland version is >= 1.13, so you should add a check for that, like in https://bugs.webkit.org/show_bug.cgi?id=161524

**Review by magomez — CHANGES_REQUESTED** [review]

File: Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp (hunk 1 above)
Comment (jacek-manko-red) [reply Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp:244]: done

File: Source/WebCore/rendering/RenderLayerCompositor.cpp (hunk 2 above)
Comment (jacek-manko-red) [reply Source/WebCore/rendering/RenderLayerCompositor.cpp:763]: done

File: Source/WebCore/rendering/RenderLayerCompositor.cpp (hunk 3 above)
Comment (jacek-manko-red) [reply Source/WebCore/rendering/RenderLayerCompositor.cpp:799]: done

File: Tools/wpe/backends/WindowViewBackend.cpp (hunk 5 above)
Comment (jacek-manko-red) [reply Tools/wpe/backends/WindowViewBackend.cpp:385]: done

File: Source/WebKit/StorageProcess/StorageProcess.cpp (hunk 4 above)
Comment (jacek-manko-red) [reply Source/WebKit/StorageProcess/StorageProcess.cpp]: done
path will only be declared if either SERVICE_WORKER or INDEXED_DATABASE are enabled.
If not, then rawWebsiteData will be marked as unused

**Review by magomez — APPROVED** [review]

**Pull request merged**